README.md
# lab3-react-native
lab3-react-native


CHANGELOG.md
## <small>2.0.1 (2019-06-20)</small>

* refactor: adopt to framework changes d80b7dc
* refactor: theme variables b9615b4
* refactor(app): common issues refactor (#137) 49f7d80, closes #137
* refactor(assets): replace remote icons with local 2f5306c
* refactor(build): replace deprecated expo imports 6ef6116
* refactor(components): missing showcases add (#133) 90fcbe0, closes #133
* refactor(components): popover showcase refactor 0b95217
* refactor(components): showcase plain implementation 5f9c1ee
* refactor(core): navigation setup ce412eb
* refactor(core): theme change (#142) 2de4a57, closes #142
* refactor(layout): auth & social 69accef
* refactor(layout): common fixes db0f315
* refactor(layout): component text style 268f382
* refactor(layout): images & keyboard 1fff7be
* refactor(layout): menu layout 69f756b
* refactor(layout): replace containers with scrollable containers 03d37af
* refactor(layout): replace fontWeight with custom fonts 9f378b0
* refactor(layout): text - eva style 4408fef
* refactor(package): adopt to eva (#144) 8b2556a, closes #144
* refactor(package): remove .idea dir c87a2a5
* refactor(package): use theme colors db9f48d
* refactor(script): travis-ci scripts a5c22c6
* refactor(theme): menus icons depending on theme choosing refactor (#143) aa4b6f7, closes #143
* docs(dev): add DEV_DOCS.md bca0ce7
* docs(readme): update README.md b91c094
* fix(app): android status bar fix a5447dd
* fix(app): change name 544a5f5
* fix(app): layouts/analytics/list-showcase  f607474
* fix(app): status bar height on android 82d1681
* fix(app): top-navigation and status bars fix (#127) 0eb19e1, closes #127
* fix(assets): restore compressed images 4bccb6e
* fix(charts): scale and reformat charts 8aea601
* fix(chat): fix baloon width (#23) 2eaaee6, closes #23
* fix(chat): fix emoji cutting on chat screens bcff510
* fix(common): add touchable for aricles4 (#27) a0c0ffd, closes #27
* fix(common): after kitten fixed fixes add (#105) bc92a4f, closes #105
* fix(common): fix expo eject (#48) 9f708ef, closes #48 #47
* fix(common): google analytics id fix a92b1d3
* fix(common): input/button-related fixes 8296333
* fix(common): lock app for portrait orientation a7c24ac
* fix(component): update container-view documentation 164d400
* fix(core): common navigation issues fix 01935e1
* fix(core): status-bar height on android 29a704d
* fix(dashboard): fix scale issues ac30e43
* fix(dashboard): set stroke color to area chart 37a7823
* fix(ecommerce): remove day from date picker on add card form eeaa67d
* fix(layout): add keyboard-avoiding behavior 549868d
* fix(layout): bouncing preventing add e4e90ad
* fix(layout): common issues 8f91da9
* fix(layout): theme values (#132) 5839a76, closes #132
* fix(layout): ui-kitten adopt 08a1e54
* fix(routes): fix walkthrough router (#21) b3c47ed, closes #21
* fix(script): first-time start 9cd370e
* fix(socials): fix avatar badges (#25) c64b40f, closes #25
* fix(theme): fix invisible status bar on android when dark theme is used b621c51
* feat(analytics): google analytics integrate (#123) 72b9e4f, closes #123
* feat(app): add splash screen (#139) c2ef6fd, closes #139
* feat(app): assets linked c48c225
* feat(app): create app navigation 3e644e4
* feat(app): project setup f12dcb6
* feat(article): article v3 add (#92) 9889b0f, closes #92
* feat(assets): dark theme icons add 6a73f6f
* feat(common): custom app-bars add (#114) 7f4d4a6, closes #114
* feat(common): make BEM an Expo app 69662cf
* feat(components): add components menu icons 082e273
* feat(components): add simple components showcase 768829b
* feat(components): interactive showcases b196f06
* feat(components): part of showcases add (#110) 0d7e288, closes #110
* feat(container): add settings container 4e965e0
* feat(dashboard): add dashboard layouts 2c60b35
* feat(e-commerce): e-commerce screens add  (#99) 77abb02, closes #99
* feat(layout): add components container d011f72
* feat(layout): add profile containers c44fa0a
* feat(layout): add send message button to chat-1 layout (#169) 50c89c8, closes #169
* feat(layout): add sign in v5 layout 540fb2d
* feat(layout): articles & feed 02fb5d6
* feat(layout): asset images & components adopt 916de74
* feat(layout): auth layouts 5fe1d4b
* feat(layout): ecommerce - rent apartment 260fe0c
* feat(layout): social - profile 137dd1d
* feat(messaging): chats add 896e8be
* feat(messaging): messages list add (#93) 684dd51, closes #93
* feat(theme): create app theme 79a2d47
* feat(theme): theme switch feature add 4ca28e8
* feat(trainings): trainings layout add (#94) cc26dee, closes #94
* build(common): add prod/dev build configurations b0e0b34
* build(common): bump react-native-ui-kitten version 8cc0a6d
* build(common): Link to react-native-ui-kitten/dev c971bbf
* build(common): setup app to watch framework and eva changes 8d196f6
* build(common): setup env configurations 9bb923d
* build(common): update android package-name 03403fb
* build(common): update app.json for expo release build. expo/react-native update (24.0.2/0.51) ec44ad7
* build(common): update eva dependencies b734930
* build(common): update eva dependencies paths 27aa0e5
* build(common): update expo to sdk-33 57e5101
* build(common): update rn/expo versions & eslint integration (#66) 00209b5, closes #66 #62 #55 #65
* build(expo): fix expo build issues f6aa6f6
* release: v2.0.0 af83498
*  fix(auth): forms values applying fix 4581e40
* build(android) Fix android build d9e971c
* build(common) removed unused npm task e0adb97
* build(common) removed unused npm task and fix react-native version bed00a9
* bump  version a97448b
* bump version ce43e7f
* bump version ad25dd3
* bump version 6067d7b
* bump version 4c8eb49
* feat(common) added interaction to social bar 3024d81
* feat(common) Added yarn lock. Navigation small fix. 5e999b6
* feat(common) Fix keyboard issues 2552c9d, closes #11 #12
* feat(data) Add db data for articles 17480db
* feat(data) Add interaction and data to login and profile screens 55df791

src/core/data/profile.ts
import {
  CategorisedProfileActivity,
  Gender,
  Profile,
  ProfileActivity,
  ProfileSocials,
} from '@src/core/model';
import {
  imageFeed1,
  imageFeed2,
  imageFeed3,
  imageFeed4,
  imageFeed5,
  imageFeed6,
  imageFeed7,
  imageFeed8,
  imageFeed9,
  imageProfile1,
  imageProfile10,
  imageProfile2,
  imageProfile3,
  imageProfile4,
  imageProfile5,
  imageProfile6,
  imageProfile7,
  imageProfile8,
  imageProfile9,
} from '@src/assets/images';

export const profile1: Profile = {
  photo: imageProfile1,
  about: 'Hola mi nombre es Jeniffer tengo 25 y vivo en berlin.' +
    ' Mis intereses son la informatica, deportes y la literatura',
  firstName: 'Jennifer',
  lastName: 'Green',
  gender: Gender.FEMALE,
  age: 25,
  weight: 48,
  height: 174,
  inseam: 45,
  email: 'jen.green@gmail.com',
  phoneNumber: '+375 44 846 97 68',
  location: 'Berlin, Alemania',
  friends: [],
  onLine: true,
};

export const profile2: Profile = {
  photo: imageProfile2,
  about: 'Hi! My name is Alexa. I\'m 25 and I live in Berlin.' +
    ' I\'m interested in computer science, music, sport and fantasy literature',
  firstName: 'Amazon',
  lastName: 'LLC',
  gender: Gender.FEMALE,
  age: 25,
  weight: 48,
  height: 174,
  inseam: 44,
  email: 'alexa.tenorio@gmail.com',
  phoneNumber: '+375 44 123 12 12',
  location: 'Portland, USA',
  friends: [],
  onLine: false,
};

export const profile3: Profile = {
  photo: imageProfile3,
  about: 'Hi! My name is Isaac. I\'m 25 and I live in Berlin.' +
    ' I\'m interested in computer science, music, sport and fantasy literature',
  firstName: 'OfficenetVE',
  lastName: 'CA',
  gender: Gender.MALE,
  age: 34,
  weight: 113,
  height: 203,
  inseam: 46,
  email: 'daly.harry@gmail.com',
  phoneNumber: '+375 44 123 12 12',
  location: 'Cleveland, USA',
  friends: [],
  onLine: true,
};

export const profile4: Profile = {
  photo: imageProfile4,
  about: 'Hi! My name is Jen. I\'m 41 and I live in Atlanta.' +
    ' I\'m interested in swag, rap.',
  firstName: 'Jen',
  lastName: 'Austin',
  gender: Gender.FEMALE,
  age: 25,
  weight: 48,
  height: 174,
  inseam: 44,
  email: 'jen.austin@gmail.com',
  phoneNumber: '+375 44 123 12 12',
  location: 'Berlin, Germany',
  friends: [profile1, profile2, profile3],
  onLine: false,
};

export const profile5: Profile = {
  photo: imageProfile5,
  about: 'Hi! My name is Kristen. I\'m 25 and I live in Berlin.' +
    ' I\'m interested in computer science, music, sport and fantasy literature',
  firstName: 'Kirsten',
  lastName: 'Crarke',
  gender: Gender.FEMALE,
  age: 25,
  weight: 48,
  height: 174,
  inseam: 44,
  email: 'kirs.clarke@gmail.com',
  phoneNumber: '+375 44 123 12 12',
  location: 'Australia',
  friends: [profile1, profile2, profile3],
  onLine: true,
};

export const profile6: Profile = {
  photo: imageProfile6,
  about: 'I\'m a Traveller. I like listening to music,' +
    ' going to the cinema, walking with my friends, drawing pictures and travelling',
  firstName: 'Helen',
  lastName: 'Kuper',
  gender: Gender.FEMALE,
  age: 25,
  weight: 48,
  height: 174,
  inseam: 44,
  email: 'helen.kuper@gmail.com',
  phoneNumber: '+375 44 123 12 12',
  location: 'Germany',
  friends: [profile1, profile2, profile3],
  onLine: false,
};

export const profile7: Profile = {
  photo: imageProfile7,
  about: 'Hi! My name is Dushane. I\'m 25 and I live in Berlin.' +
    ' I\'m interested in computer science, music, sport and fantasy literature',
  firstName: 'Dushane',
  lastName: 'Daniel',
  gender: Gender.MALE,
  age: 56,
  weight: 99,
  height: 185,
  inseam: 45,
  email: 'isaac.hunt@gmail.com',
  phoneNumber: '+375 44 123 12 12',
  location: 'NY, USA',
  friends: [profile1, profile2, profile3],
  onLine: true,
};

export const profile8: Profile = {
  photo: imageProfile8,
  about: 'Hi! My name is Shirai. I\'m 25 and I live in Berlin.' +
    ' I\'m interested in computer science, music, sport and fantasy literature',
  firstName: 'Shirai',
  lastName: 'Subaru',
  gender: Gender.FEMALE,
  age: 60,
  weight: 88,
  height: 167,
  inseam: 43,
  email: 'shirai.subaru@gmail.com',
  phoneNumber: '+375 44 123 12 12',
  location: 'Oslo, Norway',
  friends: [profile1, profile2, profile3],
  onLine: false,
};

export const profile9: Profile = {
  photo: imageProfile9,
  about: 'Hi! My name is Ray. I\'m 25 and I live in Berlin.' +
    ' I\'m interested in computer science, music, sport and fantasy literature',
  firstName: 'Ray',
  lastName: 'Cooper',
  gender: Gender.MALE,
  age: 26,
  weight: 55,
  height: 175,
  inseam: 36,
  email: 'lidmila.vilensky@gmail.com',
  phoneNumber: '+375 44 123 12 12',
  location: 'Minsk, Belarus',
  friends: [profile1, profile2, profile3],
  onLine: false,
};

export const profile10: Profile = {
  photo: imageProfile10,
  about: 'Hi! My name is Alex. I\'m 25 and I live in Berlin.' +
    ' I\'m interested in computer science, music, sport and fantasy literature',
  firstName: 'Alex',
  lastName: 'Walker',
  gender: Gender.MALE,
  age: 22,
  weight: 55,
  height: 170,
  inseam: 39,
  email: 'alex.walker@gmail.com',
  phoneNumber: '+375 44 123 12 12',
  location: 'Erlangen, Germany',
  friends: [profile1, profile2, profile3],
  onLine: false,
};

export const profileSocials1: ProfileSocials = {
  followers: 1500,
  following: 86,
  posts: 116,
};

export const profileActivity1: ProfileActivity[] = [
    {
      category: 'Plants',
      source: imageFeed1,
    },
    {
      category: 'Plants',
      source: imageFeed2,
    },
    {
      category: 'Plants',
      source: imageFeed3,
    },
    {
      category: 'Plants',
      source: imageFeed1,
    },
    {
      category: 'Plants',
      source: imageFeed2,
    },
    {
      category: 'Travel',
      source: imageFeed7,
    },
    {
      category: 'Travel',
      source: imageFeed8,
    },
    {
      category: 'Travel',
      source: imageFeed9,
    },
    {
      category: 'Travel',
      source: imageFeed8,
    },
    {
      category: 'Travel',
      source: imageFeed7,
    },
    {
      category: 'style',
      source: imageFeed4,
    },
    {
      category: 'style',
      source: imageFeed5,
    },
    {
      category: 'style',
      source: imageFeed6,
    },
    {
      category: 'style',
      source: imageFeed4,
    },
    {
      category: 'style',
      source: imageFeed5,
    },
  ]
;

export const categorisedProfileActivity1: CategorisedProfileActivity = {
  ['Plants']: [
    {
      category: 'Plants',
      source: imageFeed1,
    },
    {
      category: 'Plants',
      source: imageFeed2,
    },
    {
      category: 'Plants',
      source: imageFeed3,
    },
    {
      category: 'Plants',
      source: imageFeed1,
    },
    {
      category: 'Plants',
      source: imageFeed2,
    },
  ],
  ['Travel']: [
    {
      category: 'Travel',
      source: imageFeed7,
    },
    {
      category: 'Travel',
      source: imageFeed8,
    },
    {
      category: 'Travel',
      source: imageFeed9,
    },
    {
      category: 'Travel',
      source: imageFeed8,
    },
    {
      category: 'Travel',
      source: imageFeed7,
    },
  ],
  ['Style']: [
    {
      category: 'Style',
      source: imageFeed4,
    },
    {
      category: 'Style',
      source: imageFeed5,
    },
    {
      category: 'Style',
      source: imageFeed6,
    },
    {
      category: 'Style',
      source: imageFeed4,
    },
    {
      category: 'Style',
      source: imageFeed5,
    },
  ],
};
